src/db/api/beacon/beacon.ts

```diff
@@ -63,16 +63,24 @@ export class BeaconDB extends DatabaseService implements IBeaconDb {
 
   public async getBlock(blockRoot: bytes32): Promise<BeaconBlock> {
     const buf = await this.db.get(encodeKey(Bucket.block, blockRoot));
-    if(!buf) return null;
```

Comment:
Since only level db throws error if key not found. I've changed get implementation to catch err.notFound and return null. To make it easier to switch between db and propagating exception is not performance nor code wise very good. This change actually breaks e2e tests, they are just not well written so it says it passes

Reply:
I feel like there are several other places where we're assuming not found throws an error.
I was confused with this because the return type is still `BeaconBlock` and not `BeaconBlock | null`.
If get doesn't throw an error on not found, then we need to check all `db.getX` methods for null values?

Reply:
but otherwise we are relying everywhere in code to catch exception and check if(err.notFound) which is level specific :/

Reply:
Ok, this makes sense to me.
I think I'd like to change all the get methods to return `foo | null` and fix all the calling sites in a separate PR though, so we can be thorough and not get this one too big.